xdebug_handler_dbgp.c

```diff
@@ -2248,16 +2360,48 @@ static int xdebug_dbgp_parse_option(xdebug_con *context, char* line, int flags,
 ** Handlers for debug functions
 */
 
-static int xdebug_dbgp_cmdloop(xdebug_con *context, int bail TSRMLS_DC)
+int xdebug_dbgp_cmdloop(xdebug_con *context, int bail TSRMLS_DC, int block)
 {
 	char *option;
-	int   ret;
+	int    ret;
+	int    length;
 	xdebug_xml_node *response;
 
+#if WIN32|WINNT
+	u_long                     yes = 1;
+	u_long                     no = 0;
+	int                        status;
+#endif
+
 	do {
-		option = xdebug_fd_read_line_delim(context->socket, context->buffer, FD_RL_SOCKET, '\0', NULL);
-		if (!option) {
-			return 0;
+		option=NULL;
+		length=0;
+		// set to block mode
+		if(!block){
+#ifdef WIN32
+			status = ioctlsocket(context->socket, FIONBIO, &yes);
+			if (SOCKET_ERROR == status) {
+				XG(context).handler->log(XDEBUG_LOG_WARN, "setting socket to FIONBIO: %d.\n", WSAGetLastError());
+			}
+#else
+			fcntl(context->socket, F_SETFL, O_NONBLOCK);
+#endif
+		}
+		ret = xdebug_fd_read_line_delim(context->socket, context->buffer, FD_RL_SOCKET, '\0', &option, &length, block);
+		// set it back
+		if(!block){
+#ifdef WIN32
+			status = ioctlsocket(context->socket, FIONBIO, &no);
+			if (SOCKET_ERROR == status) {
+				XG(context).handler->log(XDEBUG_LOG_WARN, "setting socket not to FIONBIO: %d.\n", WSAGetLastError());
+			}
+#else
+			int flags = fcntl(context->socket, F_GETFL, 0);
+			fcntl(context->socket,F_SETFL,flags&~O_NONBLOCK);
+#endif
+		}
+		if (ret || length<=0) {
+			return ret;
```

Comment:
Please go through this block and fix all the whitespace issues: always a space after a control statement (such as `if`), always spaces after a `,`, and always spaces around comparisons ans assignments such as `&`, and `<=`.